webapp/src/components/user_selector/autocomplete_selector.tsx

```diff
@@ -0,0 +1,126 @@
+        neutral10: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.16),
+        neutral20: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.24),
+        neutral30: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.32),
+        neutral40: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.40),
+        neutral50: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.56),
+        neutral60: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.64),
+        neutral70: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.72),
+        neutral80: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.8),
+        neutral90: blendColors(theme.centerChannelBg, theme.centerChannelColor, 0.88),
+    };
+});
+
+export default function AutocompleteSelector(props: Props) {
+    const {
+        loadOptions,
+        label,
+        labelClassName,
+        helpText,
+        inputClassName,
+        placeholder,
+        disabled,
+        onSelected,
+        theme,
+    } = props;
+
+    const handleSelected = (selected: ValueType<UserProfile>) => {
+        if (onSelected) {
+            onSelected(selected);
+        }
+    };
+
+    let labelContent;
+    if (label) {
+        labelContent = (
+            <label
+                className={'control-label ' + labelClassName}
+            >
+                {label}
+            </label>
+        );
+    }
+
+    let helpTextContent;
+    if (helpText) {
+        helpTextContent = (
+            <div className='help-text'>
+                {helpText}
+            </div>
+        );
+    }
+
+    const getOptionData = (option: UserProfile) => option.username;
+
+    const colors = getColorStyles(theme);
+
+    return (
+        <div
+            data-testid='autoCompleteSelector'
+            className='form-group'
+        >
+            {labelContent}
+            <div className={inputClassName}>
+                <AsyncSelect
+                    cacheOptions={true}
+                    loadOptions={loadOptions}
+                    defaultOptions={true}
+                    isClearable={true}
+                    disabled={disabled}
+                    placeholder={placeholder}
+                    getOptionLabel={getOptionData}
+                    getOptionValue={getOptionData}
+                    onChange={handleSelected}
+                    styles={{
+                        option: (provided, state) => ({
+                            ...provided,
+                            color: state.isDisabled ? colors.neutral30 : colors.neutral90,
+                        }),
+                    }}
+                    theme={(componentTheme) => ({
+                        ...componentTheme,
+                        colors: {
+                            ...componentTheme.colors,
+                            ...colors,
+                        },
```

Comment:
This seems a bit complicated to have this deep in the render method. Can this be done earlier in this function, or possibly in a new hook, like [useTheme](https://reactjs.org/docs/hooks-custom.html)? This would include computing the `styles` props as well.